### PR TITLE
test: raise coverage for music search component

### LINKED ISSUE #363: Raise Coverage For Music Search Component (>3%)
⚠️Before implementing this task , check if Music Search Components is already merged in develop branch
[](
https://github.com/Liatoshynsky-Foundation/lf-client/pull/342)

**Description:**
Some standalone functions or utilities currently have less than 3% test coverage, which makes them vulnerable to bugs and regressions. These functions are likely not hit by existing tests and may include edge case logic, formatting, calculations, or conditionals that require more reliable validation.

Image

### PR DESCRIPTION
**Pull request by Kryzhanivsky**
## Summary of change

- Added tests for the MusicSearch component.
- Test coverage was increased.

Screenshot:
image

### CODE REVIEW

**Review by oleg191006 — APPROVED** [review]

**Review by ssashayurchenko — COMMENTED** [review]
Maybe using userEvent instead of fireEvent would be a better choice for interacting with user input fields, such as a search input?

File: app/shared/components/composition-search/MusicSearch.test.tsx [hunk 1]
```diff
@@ -42,23 +45,26 @@ describe('MusicSearch component', () => {
...
   beforeEach(() => {
     render(<MusicSearch onFilterChange={handleChange} data={mockMusicData} />);
   });
 
   it('should render the MusicSearch icon', () => {
     expect(screen.getByAltText('search')).toBeInTheDocument();
   });
-  it('should render the Autcomplete and check list opening and closing', async () => {
+
+  it('should render the Autocomplete and check list opening and closing', async () => {
```
Comment (Iarynovskyi) [inline app/shared/components/composition-search/MusicSearch.test.tsx]:
```suggestion
  it('should render the Autocomplete and check list opening and closing', () => {
```
You delete `await` in 63 line so now `async` here is unused
So delete it here too, please

File: app/shared/components/composition-search/MusicSearch.test.tsx [hunk 2]
```diff
@@ -70,4 +76,64 @@ describe('MusicSearch component', () => {
...
+
+  it('should render no options text when no data is provided', () => {
+    cleanup();
+    render(<MusicSearch onFilterChange={handleChange} data={[]} />);
+    const input = screen.getByRole('combobox');
+    fireEvent.mouseDown(input);
+    expect(screen.getByText('Не знайдено')).toBeInTheDocument();
+  });
+
+  it('should debounce onFilterChange calls', async () => {
```
Comment (Iarynovskyi) [inline app/shared/components/composition-search/MusicSearch.test.tsx]:
```suggestion
  it('should debounce onFilterChange calls', () => {
```

**Review by Iarynovskyi — APPROVED** [review]

**Review by ssashayurchenko — APPROVED** [review]

**Review by luvthenika — APPROVED** [review]
